### PR TITLE
Rosetta: /construction/derive endpoint

### PR DESCRIPTION
**Pull request by asimm241**
- [ ] Rosetta construction/derive endpoint

### CODE REVIEW

File: src/rosetta-helpers.ts [hunk 1]
```diff
@@ -169,3 +172,17 @@ function makePoisonMicroblockOperation(tx: DbMempoolTx | DbTx, index: number): R
 
   return sender;
 }
+
+export function publicKeyToBitcoinAddress(publicKey: string): string {
+  const publicKeyBuffer = Buffer.from(publicKey, 'hex');
+
+  const address = btc.payments.p2pkh({
+    pubkey: publicKeyBuffer,
+    network: btc.networks.regtest,
```
Comment (zone117x) [inline src/rosetta-helpers.ts:181]: This can't be hardcoded to regtest.

Comment (asimm241) [reply src/rosetta-helpers.ts]: I have updated the PR with this fix.

File: src/rosetta-helpers.ts [hunk 2]
```diff
@@ -169,3 +173,24 @@ function makePoisonMicroblockOperation(tx: DbMempoolTx | DbTx, index: number): R
...
+    btcNetwork = btc.networks.bitcoin;
+  } else {
+    btcNetwork = btc.networks.regtest;
+  }
+
+  const address = btc.payments.p2pkh({
+    pubkey: publicKeyBuffer,
+    network: btcNetwork,
+  });
+  return address.address ? address.address : '';
```
Comment (zone117x) [inline src/rosetta-helpers.ts:191]: This is a failure condition correct? If so, it should throw rather than fail silently.

Comment (asimm241) [reply src/rosetta-helpers.ts]: This function is being called in a try-catch block that will handle if `p2pkh` fails. That being said this function should be autonomous. I will fix and update the PR.

**Review by zone117x — APPROVED** [review]
LGTM!

**Pull request merged**